starter-code/database.js

```diff
@@ -134,7 +133,13 @@ class Database {
         // Think if you may need to implement two queries chained
         // remeber once it's finish to comment callback('Error buying product');
         
-        callback('Error buying product');
+        // let product = database.collection('products').find({'name':productName})
+        // database.collection('users').update({'firstName':userFirstName}, {$push: {"shoppingCart": product}})
+        
+        database.collection('products').findOne({'name': productName}, (error, result) => {
+          database.collection('users').update({'firstName':userFirstName}, {$push: {"shoppingCart": result}})
+          callback(error, result)
```

Comment:
You actually want to execute the `callback` function as a callback of your update query. Otherwise you are returning before knowing if the request has actually been done.